ci-operator/step-registry/confidential-clusters/ref/cocl/beaker-kind-provision/confidential-clusters-ref-cocl-beaker-kind-provision-commands.sh

```diff
@@ -0,0 +1,1283 @@
+  echo "[ERROR] Could not find extracted directory"
+  echo "[INFO] Listing all directories in ${HOME}:"
+  ls -la "${HOME}"
+  exit 1
+fi
+
+echo "[INFO] Found extracted directory: ${EXTRACTED_DIR}"
+
+# Rename extracted directory to expected name
+# Note: WORK_DIR must not exist for mv to rename (not create WORK_DIR as parent)
+mv "${EXTRACTED_DIR}" "${WORK_DIR}"
+rm -f "${TEMP_TARBALL}"
+
+echo "[INFO] Directory structure after rename:"
+ls -la "${WORK_DIR}" | head -10
+
+cd "${WORK_DIR}"
+echo "[SUCCESS] Repository downloaded and extracted to ${WORK_DIR}"
+
+# ============================================================================
+# Adapt Kind Config for External Access
+# ============================================================================
+echo "[INFO] Adapting kind configuration for external access..."
+
+# Backup original config
+cp kind/config.yaml kind/config.yaml.orig
+
+# Add API server configuration for external access
+cat > kind/config.yaml << KINDCONFIG
+# Adapted from cocl-operator kind/config.yaml for external access
+kind: Cluster
+apiVersion: kind.x-k8s.io/v1alpha4
+networking:
+  apiServerAddress: "${BEAKER_IP}"
+  apiServerPort: 6443
+containerdConfigPatches:
+- |-
+  [plugins."io.containerd.grpc.v1.cri".registry]
+    config_path = "/etc/containerd/certs.d"
+nodes:
+- role: control-plane
+  extraPortMappings:
+  # Note: 6443 is already handled by apiServerAddress/apiServerPort above
+  # Adding it here would cause "address already in use" error
+  - containerPort: 31000
+    hostPort: 8080
+    protocol: TCP
+  - containerPort: 31001
+    hostPort: 8000
+    protocol: TCP
+featureGates:
+  "ImageVolume": true
+KINDCONFIG
+
+echo "[INFO] Adapted kind configuration:"
+cat kind/config.yaml
+
+# ============================================================================
+# Patch scripts/common.sh to fix DOCKER_HOST for root docker
+# ============================================================================
+echo "[INFO] Patching scripts/common.sh to fix DOCKER_HOST..."
+echo "[INFO] Current directory: $(pwd)"
+echo "[INFO] Searching for scripts/common.sh..."
+find . -name "common.sh" -type f 2>/dev/null | head -5 || echo "  No common.sh found"
+
+if [ -f "scripts/common.sh" ]; then
+  echo "[INFO] Found scripts/common.sh, backing up and rewriting..."
+
+  # Backup original
+  cp scripts/common.sh scripts/common.sh.orig
+
+  echo "=========================================="
+  echo "[DEBUG] Original scripts/common.sh content:"
+  cat -n scripts/common.sh
+  echo "=========================================="
+
+  # Don't try to patch with sed - just completely rewrite it
+  # This avoids bash syntax errors from empty if blocks
+  echo "[INFO] Completely rewriting scripts/common.sh for root docker..."
+
+  cat > scripts/common.sh << 'COMMON_SH_FIXED'
+#!/bin/bash
+# Patched by CI to fix DOCKER_HOST issue
+# Root docker uses /var/run/docker.sock by default, no need to set DOCKER_HOST
+
+RUNTIME=${RUNTIME:-docker}
```

Comment:
Right now, the kind cluster only works for docker, we still needs to fix the root podman configuration. Please, just set docker as RUNTIME and remove all the logic for podman